devops/jobs/ReplaceUsernames.groovy

```diff
@@ -0,0 +1,97 @@
+package devops.jobs
+
+/* JenkinsPublicConstants can be found in src/main/groovy/org/edx/jenkins/dsl*/
+/* It contains values and methods that are used repeatedly through the dsl jobs */
+import org.yaml.snakeyaml.Yaml
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_WORKER
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.GENERAL_SLACK_STATUS
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_GITHUB_BASEURL
+
+/*
+Need to manually add files with the following details to $WORKSPACE/username-replacement-secure/$ENVIRONMENT.yml:
+    client_id: '<secret>'
+    client_key: '<secret>'
+    base_urls:
+        lms: http://localhost:18000
+        ecommerce: http://localhost:18130
+        discovery: http://localhost:18381
+        credentials: http://localhost:18150
+
+*/
+class ReplaceUsernames {
+    public static def job = { dslFactory, extraVars ->
+        assert extraVars.containsKey('ENVIRONMENTS') : "Please define ENVIRONMENTS. It should be a list of strings."
+        assert !(extraVars.get('ENVIRONMENTS') instanceof String) : "Make sure ENVIRONMENTS is a list and not a string"
+
+        extraVars.get('ENVIRONMENTS').each { environment ->
+            dslFactory.job("${environment}-replace-usernames") {
+                /* TODO: Find out if this is correct */
+                /* Enable project based security */
+                authorization {
```

Comment:
Security is handled at the folder level so you don't need this block at all.

Reply:
Removed